### PR TITLE
feature (addon): #414 context menu share

### LINKED ISSUE #414: Implement context menu share
This ticket should implement a share action in the existing context menu on New Tab/Timeline items. This will require add-on apis as well as new UI in the `LinkMenu` component. It should use the same strategy/options that are currently in the share plane (this will be iterated on later).

[Edit by @k88hudson]

> Currently you can only delete Top Sites and Highlighted sites.
But for Top/Recent Activity and Bookmarks, you can share and do "...".
Ideally we should be able to share Highlighted results since they're presumably "important" content.

> firefox_start_and_firefox_start_and_new_tab

> firefox_start_and_firefox_start_and_new_tab

> UX Ref: http://people.mozilla.org/~shorlander/activity-stream/activity-stream-latest/start-page.html

### PR DESCRIPTION
**Pull request by rlr**
@k88hudson r?

Fixes #414

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/1186)

### CODE REVIEW

File: addon/ShareProvider.js [hunk 1]
```diff
@@ -25,104 +25,20 @@ try {
   SocialService = Cu.import("resource:///modules/SocialService.jsm", {}).SocialService;
 }
```
Comment (rlr) [inline addon/ShareProvider.js]: The changes in this file are just moving things around so I could reuse the same share code being used by the share toolbar. The diff makes it look like a lot changed, but the actual sharing code didn't change.

File: content-src/components/ContextMenu/ContextMenu.scss [hunk 2]
```diff
@@ -33,8 +34,33 @@
...
   }
 }
+
+.submenu {
+  position: relative;
+
+  .context-menu {
+    display: none;
+    left: auto;
+    right: 95%;
```
Comment (rlr) [inline content-src/components/ContextMenu/ContextMenu.scss]: Right now the share sub menu always opens to the left. I would love it to open to the right when we know there is enough space. But I have no idea how to do that. Help?

File: content-src/lib/fake-data.js [hunk 3]
```diff
@@ -44,5 +44,13 @@ module.exports = {
   "Prefs": {
     "prefs": {},
     "error": false
+  },
+  "ShareProviders": {
+    "providers": [
+      {
+        "name": "MySpace",
```
Comment (rlr) [inline content-src/lib/fake-data.js]: get off my lawn. lol

**Comment by k88hudson** [comment]
Waiting until next week to review this

**Comment by rlr** [comment]
@k88hudson I'm gonna rebase this and let you know when done so you can r?. crossing fingers for not many conflicts.

File: addon/ShareProvider.js [hunk 4]
```diff
@@ -210,10 +126,140 @@ ShareProvider.prototype = {
...
     this._createButton();
-    Overlay.init(this.eventTracker);
+    Overlay.init(this);
   }),
 
   /**
+   * Return the installed social share providers.
+   */
+  get socialProviders() {
+    const defaultSort = ["Facebook", "Twitter", "Tumblr", "LinkedIn", "Yahoo Mail", "Gmail"];
```
Comment (k88hudson) [inline addon/ShareProvider.js]: maybe store these as a global constant? if not in `common/constants` at least up top?

File: content-src/components/ContextMenu/ContextMenu.scss (hunk 2 above)
Comment (k88hudson) [inline content-src/components/ContextMenu/ContextMenu.scss]: You have to do it by reading the position of the element relative to the window :S definitely a follow-up...

**Comment by k88hudson** [comment]
@rlr looks mostly good to me, just needs a rebase!

**Comment by rlr** [comment]
rebased and feedbacked ^

**Pull request merged**